clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/security/DcosCredentialMap.groovy

```diff
@@ -0,0 +1,19 @@
+package com.netflix.spinnaker.clouddriver.dcos.security
+
+class DcosCredentialMap {
+    final List<DcosClusterCredentials> dcosClusterCredentials
```

Comment:
Instead of a list, use a map here. Then getCredentialsByCluster doesn't have to search the list each time to retrieve by cluster name, and getCredentials() can just use map.values().